Finishes Bowling Test Generation

bmeverett opened the pull request:
Finishes up #470 
Closes #402 

The `RenderTestMethodBodyAct` is a little messier then I'd like, but it gets the job done.

ErikSchierboom commented on exercises/bowling/BowlingTest.cs:
```diff
@@ -1,207 +1,303 @@
+        DoRoll(previousRolls, sut);
+        sut.Roll(-1);
+        var actual = sut.Score();
+        Assert.Null(actual);
```
I would prefer the methods that have erroneous input (see the [canonical-data](https://github.com/exercism/problem-specifications/blob/master/exercises/bowling/canonical-data.json#L133)) to throw an exception, instead of returning `null`.

ErikSchierboom commented on generators/Exercises/Bowling.cs:
```diff
@@ -0,0 +1,87 @@
+            {
+                if (!(canonicalDataCase.Expected is long))
+                {
+                    canonicalDataCase.Expected = null;
```
See comment above, I would like to throw an exception in this case.

ErikSchierboom commented on generators/Exercises/Bowling.cs:
```diff
@@ -0,0 +1,87 @@
+                }
+                canonicalDataCase.SetInputParameters();
+                canonicalDataCase.UseVariableForTested = true;
+                canonicalDataCase.Property = "score";
```
When we switch to using exceptions, this line should be removed as the actual property being tested is then not `score`, but `roll`.

ErikSchierboom commented:
@bmeverett By the way, the older [PR](https://github.com/exercism/csharp/pull/470/files) for this generator shows you how to use exceptions.

bmeverett commented:
@ErikSchierboom I implemented your fixes. I changed the test cases to throw exceptions, and in the case of roll I implemented the function to check for valid input so that it would throw exceptions for invalid input.

ErikSchierboom commented:
@bmeverett Looking great! I'll wait for CI to become green and then I'll merge.

ErikSchierboom commented:
Merged. Thanks! 🎉
